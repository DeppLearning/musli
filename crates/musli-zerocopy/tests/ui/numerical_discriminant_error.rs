//! This ensure that the `ZeroCopy` macro produces decent diagnostics when
//! dealing with simple numerical discriminants.

use musli_zerocopy::ZeroCopy;

#[derive(ZeroCopy)]
#[repr(u8)]
enum UnsignedNegative {
    Variant = 1u8
}

#[derive(ZeroCopy)]
#[repr(u8)]
enum Overflow {
    Variant = 255,
    Variant2
}

fn main() {
}
