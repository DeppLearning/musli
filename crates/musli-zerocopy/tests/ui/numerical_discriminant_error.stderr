error: Discriminant overflow for representation
  --> tests/ui/numerical_discriminant_error.rs:15:15
   |
15 |     Variant = 255,
   |               ^^^

error[E0370]: enum discriminant overflowed
  --> tests/ui/numerical_discriminant_error.rs:16:5
   |
16 |     Variant2
   |     ^^^^^^^^ overflowed on value after 255
   |
   = note: explicitly set `Variant2 = 0` if that is desired outcome
